framework/src/st_things/things_stack/utils/things_ping.c

```diff
@@ -116,172 +269,244 @@ bool things_ping_init(void)
 	} else {
 		res = true;
 	}
+	// Unlock the mutex.
 	pthread_mutex_unlock(&mutex_ping_list);
 
 	p_ping = NULL;
 	THINGS_LOG_D(TAG, "Exit.");
 	return res;
 }
 
+//-------------------------------------------------------------------------------
+/**
+ * Function definition for terminate OIC device keepAlive for ping resource.
+ */
 bool things_ping_terminate(void)
 {
 	THINGS_LOG_D(TAG, "Enter.");
 
+	// Lock the mutex.
 	pthread_mutex_lock(&mutex_ping_list);
 	if (list) {
 		THINGS_LOG_D(TAG, "things_ping_s terminate.");
 		things_ping_s *ping = NULL;
 		while ((ping = (things_ping_s *) list->pop(list)) != NULL) {
 			THINGS_LOG_D(TAG, "Terminate ping.(%s)", ping->addr);
 			if (cas_mask(ping, PING_ST_STARTTHREAD, true, PING_ST_INIT) == true) {
+				// Destory ping thread.
 				things_ping_destroy_thread(ping);
 			}
+			// Terminate the ping.
 			terminate_things_ping_s(ping);
 		}
 
 		THINGS_LOG_D(TAG, "Terminate ping-list.");
+		// Terminate the queue.
 		terminate_list(list);
 		list = NULL;
 	}
 
+	// Unlock the mutex.
 	pthread_mutex_unlock(&mutex_ping_list);
 	pthread_mutex_destroy(&mutex_ping_list);
 
 	THINGS_LOG_D(TAG, "Exit.");
 	return true;
```

Comment:
if always return will true,
i think better to change return type is void.

Reply:
changed.